### ISSUE TITLE
Extend input PDF's Catalog

### ISSUE THREAD

**Issue by daleblay0** [J-F-Liu/lopdf#165]
How can I use lopdf to modify an input PDF file's `Catalog`?

The original `Catalog` is something like:

`6 0 obj<</Type/Catalog/Pages 1 0 R>>endobj`

I'd like to modify this to be something like:

`6 0 obj<</Type/Catalog/Pages 1 0 R/KEY/VALUE>>endobj`

I've managed to get close with the below:

```
        let mut doc = Document::load("example.pdf").unwrap();
        doc.version = "1.4".to_string();

        for (object_id, object) in doc.objects.clone() {
            match object.type_name().unwrap_or("") {
                "Catalog" => {
                    let catalog = doc.get_object_mut(object_id).unwrap();
                    let catalog_dict = catalog.as_dict_mut().unwrap();

                    let mut other_dict = dictionary!{};
                    for (k, v) in catalog_dict.iter() {
                        other_dict.set(k.clone(), v.clone());
                    }

                    other_dict.set("KEY", "VALUE");

                    catalog_dict.extend(&other_dict);
                },
                _ => ()
            }
        }

        doc.save("updated.pdf").unwrap();
```

But, the output is wrong:

`6 0 obj<</Type/Catalog/Pages[1 0 R 1 0 R]/KEY/VALUE>>endobj`

**Comment by fold-squirrel** [comment]
don't use extend, instead just make an object from the "other" dictionary and replace it.
`doc.objects.insert(object_id, Object::Dictionary(other_dict));`

**Comment by Heinenen** [comment]
I agree that lopdf's behavior here is not what what one would expect.
I would say the problem here is that lopdf's `extend` works recursively which clashes with the definition of `std::iter::Extend`.

A sensible approach could be to implement an `extend` function that matches the std. If the recursive extend from lopdf can't just be thrown out, it could be renamed to `extend_recursive`.

**Comment by Heinenen** [comment]
As for the actual question, you can just call `catalog_dict.set(KEY, VALUE)`, multiple times if needed.
